micrometer-observation-test/src/main/java/io/micrometer/observation/tck/ObservationContextAssert.java

```diff
@@ -453,8 +453,7 @@ private ObservationView checkedParentObservation() {
         if (p == null) {
             failWithMessage("Observation should have a parent");
         }
-        // see: https://github.com/uber/NullAway/issues/1222
-        return Objects.requireNonNull(p);
+        return p;
```

Comment:
Although NullAway is now happy, I still see a warning from IntelliJ:

<img width="1067" height="741" alt="Image" src="https://github.com/user-attachments/assets/62fb9d4c-e571-4e4e-9d87-9bf2afb6377d" />

I'm not sure if there is anything else that I can do here.

Is IntelliJ detecting only `org.jetbrains.annotations.@Contract`?

Reply:
It may be addressed by IntelliJ 2025.2. According to [this](https://www.jetbrains.com/idea/whatsnew/#page__content-jspecify-support--a-major-step-toward-safer-code), support for jSpecify was added.

Reply:
I just upgraded but it's still the same, unfortunately, which kind of makes sense because custom `@Contract` annotations are unrelated to JSpecify, [at least for now](https://github.com/uber/NullAway/issues/1222#issuecomment-3113055812).

Reply:
Usually it's the other way :) but I guess this is an issue we need to live with for now.